runtime/runtime_impl.hpp

```diff
@@ -64,22 +70,35 @@ class runtime_impl : public runtime {
 	}
 
 	virtual void wait() override {
-		while (!terminate.load()) {
+		while (!pb.lookup_impl<Stoplight>()->should_stop()) {
```

Comment:
This loop is no longer required because of the blocking wait below. `wait_for_shutdown()` can only ever return if `should_stop()` is true, so this check is redundant.
The comment below might need a slight tweak without the presence of this loop.

Reply:
Agree, will change